### PR TITLE
[Go Hack] Go version upgrade and code cleanup

### PR DESCRIPTION
**Pull request by pedroldk**
The objective of this PR is to

- Upgrade Go version (which will allow taking advantage of generics and keep the Go version up to date)
- Some code cleanup

### CODE REVIEW

**Comment by pedroldk** [comment]
Hi @reguero, just a tiny contribution if you want to review it please 😉

**Comment by reguero** [comment]
Hi @pedroldk 
Many thanks!
Please note that we are not considering generics yet.
Having a look in any case.

**Comment by dolmen** [comment]
A single commit for unrelated changes. @pedroldk This is not a good practice.

File: lbcluster/lbcluster.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ func (lbc *LBCluster) concatenateIps(myIps []net.IP) string {
 	return strings.Join(ip_string, " ")
 }
 
-//Find_best_hosts Looks for the best hosts for a cluster
+// FindBestHosts Find_best_hosts Looks for the best hosts for a cluster
```
Comment (dolmen) [inline lbcluster/lbcluster.go:135]: The documentation line is still incorrect.

File: lbd.go [hunk 2]
```diff
@@ -204,8 +203,6 @@ func main() {
 			lg.Error("Got an unexpected value")
 		}
 	}
-	lg.Error("The lbd is not supposed to stop")
```
Comment (dolmen) [inline lbd.go:207]: Why is this line removed?

This should have been explained in the commit message. But as the commit bundles many things, this change is hidden.

**Comment by pedroldk** [comment]
> A single commit for unrelated changes. @pedroldk This is not a good practice.

Not really unrelated because as you can see in the PR title, this is about code cleanup and Go version upgrade. Going to drop this PR for now.

**Comment by dolmen** [comment]
"Code cleanup" and "Go version upgrade" are unrelated changes. They should have been submitted as 2 different PR to allow the maintainers to selectively accept one independently of the other.

File: tests/get_state_dns_test.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ func TestGetStateDNS(t *testing.T) {
 	}
 	//receiving the output for every alias and storing the results into a map
 	received := make(map[string][]interface{})
-	iprecString := []string{}
```
Comment (dolmen) [inline tests/get_state_dns_test.go:32]: @pedroldk Proposing this change as "code cleanup" is misleading because it might introduce bugs. The original code initialize the slice to a zero element array, while the new one initialize to `nil`.

The difference might not matter in this code (this is just a test file anyway), but advertising this change as just "code cleanup" is wrong.

**Comment by pedroldk** [comment]
> "Code cleanup" and "Go version upgrade" are unrelated changes. They should have been submitted as 2 different PR to allow the maintainers to selectively accept one independently of the other.

You are correct, since the change was so small I did not see the point in it but it's better to separate waters as you say 👍  https://github.com/cernops/golbd/pull/14
